libcxx/include/__algorithm/ranges_clamp.h

```diff
@@ -37,9 +37,10 @@ struct __fn {
     _LIBCPP_ASSERT_UNCATEGORIZED(!bool(std::invoke(__comp, std::invoke(__proj, __high), std::invoke(__proj, __low))),
                                  "Bad bounds passed to std::ranges::clamp");
 
-    if (std::invoke(__comp, std::invoke(__proj, __value), std::invoke(__proj, __low)))
+    auto&& __projected = std::invoke(__proj, __value);
```

Comment:
This could dangle.

Reply:
I don't see how it could dangle. Do you mean that it will dangle if `std::invoke(__proj, __value)` returns a temporary? Rvalue references extend the lifetime of temporaries so whatever this references will live until the end of the function.

Reply:
That's not quite true. It will lifetime extend any _value_ returned from that function. However if the function returns a reference to a temporary created in the call to the projection, then that reference will dangle once at the semicolon.

For example, consider the identity projection that tries to prevent copies:

```
std::string const& identity_proj(std::string const& x) { return x; }
```
No matter what,  the return value ends its lifetime when the argument does. If you pass something other than `std::string` to that function, you're in trouble. 

Now, it may be the case that any such code is buggy. I'm not 100% convinced it is. I haven't found standardese that clarifies the situation. But it would work today, and we could continue to make it work my forwarding the projected values down to another function which uses them.

Reply:
The `invoke(__proj, __value)` call itself creates no temporaries; it just binds two references. Any temporaries are created inside `invoke` and are destroyed before `invoke` returns, so `invoke(identity_proj, "")` is always dangling regardless of what you do with the result.